woocommerce/woocommerce
# Open variation preview in a new tab

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the code to open the variation preview in a new tab.

**Acceptance criteria**

- [ ] Clicking Preview in the quick actions menu opens the preview in a new tab.

Closes #40173.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that the `New product editor` feature is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Ensure that the `product-variation-management` feature is enabled under the `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper`.
3. Navigate to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and click the `Add variation options` button from the `Variations` tab.
4. After creating the variations, click the three dots icon in the line of the first variation and select `Preview`.
5. This action should open a new tab with the selected variation.

![Screen Capture on 2023-09-25 at 16-16-21](https://github.com/woocommerce/woocommerce/assets/1314156/cd1b2bd2-a372-4358-a64d-8a372a1d095e)



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
